djangoProject/users/views.py

```diff
@@ -62,4 +58,38 @@ def post(self, request):
             return Response(
                 {"error": str(e)},
                 status=status.HTTP_400_BAD_REQUEST,
-            )
\ No newline at end of file
+            )
+
+
+class SignInView(APIView):
+    permission_classes = [AllowAny]
+
+    def post(self, request):
+        email = request.data.get('email')
+        password = request.data.get('password')
+
+        if not email or not password:
+            return Response(
+                {"error": "Email and password are required."},
+                status=status.HTTP_400_BAD_REQUEST
+            )
```

Comment:
Returning the same error message (`{"error": "Invalid email or password."}`) for all failures could reveal information about valid emails in the system.
Use a generic error message to avoid information leakage:
```
return Response(
    {"error": "Invalid credentials."},
    status=status.HTTP_401_UNAUTHORIZED
)
```